### PR TITLE
[NEW] Unread count on header back button

### LINKED ISSUE #778: [NEW] Show unread count in room's back button

### PR DESCRIPTION
**Pull request by IlarionHalushka**
@RocketChat/ReactNative
Closes #778 

This PR aims to add an "unread messages counter" inside of RoomView next to the header back button.

Screenshots:
Screen Shot 2019-08-07 at 10 53 46 PM
Screen Shot 2019-08-07 at 10 54 20 PM

### CODE REVIEW

File: app/views/RoomView/index.js [hunk 1]
```diff
@@ -99,6 +98,19 @@ export default class RoomView extends React.Component {
 					navigation={navigation}
 					toggleFollowThread={toggleFollowThread}
 				/>
+			),
+			headerLeft: (
+				<View onLayout={(event) => {
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: onLayout is needed?

File: yarn.lock [hunk 2]
```diff
@@ -3509,6 +3509,11 @@ balanced-match@^1.0.0:
   resolved "https://registry.yarnpkg.com/balanced-match/-/balanced-match-1.0.0.tgz#89b4d199ab2bee49de164ea02b89ce462d71b767"
   integrity sha1-ibTRmasr7kneFk6gK4nORi1xt2c=
 
+base-64@0.1.0:
```
Comment (djorkaeffalexandre) [inline yarn.lock]: why?

File: app/views/RoomView/index.js [hunk 3]
```diff
@@ -99,6 +98,19 @@ export default class RoomView extends React.Component {
...
+					console.warn(event.nativeEvent.layout);
+				}}
+				>
+					<HeaderBackButton
+						title={unreadsCount > 999 ? '+999' : unreadsCount}
+						backTitleVisible
+						onPress={() => navigation.goBack()}
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: `navigation.goBack`

File: app/views/RoomView/index.js [hunk 4]
```diff
@@ -79,16 +79,15 @@ export default class RoomView extends React.Component {
 		const t = navigation.getParam('t');
 		const tmid = navigation.getParam('tmid');
 		const toggleFollowThread = navigation.getParam('toggleFollowThread', () => {});
+		const unreadsCount = navigation.getParam('unreadsCount') > 0 ? navigation.getParam('unreadsCount') : null;
```
Comment (diegolmello) [inline app/views/RoomView/index.js]:
```jsx
const unreadsCount = navigation.getParam('unreadsCount', null);
```

File: app/views/RoomView/index.js (hunk 1 above)
Comment (diegolmello) [reply app/views/RoomView/index.js]: Not needed indeed

**Comment by djorkaeffalexandre** [comment]
@IlarionHalushka can you add Android Screenshots and update iOS Screenshots?

**Comment by djorkaeffalexandre** [comment]
@IlarionHalushka I think we not need `back` string:

File: app/views/RoomView/index.js [hunk 5]
```diff
@@ -72,6 +71,14 @@ class RoomView extends React.Component {
 					navigation={navigation}
 					toggleFollowThread={toggleFollowThread}
 				/>
+			),
+			headerLeft: (
+				<HeaderBackButton
+					title={unreadsCount > 999 ? '+999' : unreadsCount}
+					backTitleVisible
+					onPress={navigation.goBack}
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: I don't understand why, but this not work on Android.. only `() => navigation.goBack()`

File: app/views/RoomView/Header/Header.js [hunk 6]
```diff
@@ -106,7 +108,7 @@ const Header = React.memo(({
 	}
 
 	return (
-		<View style={[styles.container, { width: width - widthOffset }]}>
+		<View style={[styles.container, { width: width - (width * tmid ? 0.2 : 0.3) }]}>
```
Comment (diegolmello) [inline app/views/RoomView/Header/Header.js]: Why did you remove `widthOffset`?
What `width * tmid` does?

This way, when `unread` is null, `title` is not left aligned nor centered.



This is what widthOffset does:

File: app/views/RoomView/index.js [hunk 7]
```diff
@@ -72,6 +71,14 @@ class RoomView extends React.Component {
 					navigation={navigation}
 					toggleFollowThread={toggleFollowThread}
 				/>
+			),
+			headerLeft: (
+				<HeaderBackButton
+					title={unreadsCount > 999 ? '+999' : unreadsCount || ' '}
+					backTitleVisible
+					onPress={() => navigation.goBack()} // on Android onPress={navigation.goBack} doesn't work
```
Comment (diegolmello) [inline app/views/RoomView/index.js]: You can remove this comment.
